app/controllers/ChoiceController.scala

```diff
@@ -35,14 +35,14 @@ class ChoiceController @Inject()(authenticate: AuthAction, cache: CacheRepositor
   implicit ec: ExecutionContext
 ) extends FrontendController(mcc) with I18nSupport {
 
-  def displayChoiceForm(): Action[AnyContent] = authenticate.async { implicit request =>
-    cache
-      .findByEori(request.eori)
-      .map(_.map(cache => Choice(cache.answers.`type`)))
-      .map {
-        case Some(choice) => Ok(choicePage(Choice.form().fill(choice)))
-        case None         => Ok(choicePage(Choice.form()))
-      }
+  def displayChoiceForm: Action[AnyContent] = authenticate.async { implicit request =>
+    cache.findByEori(request.eori).map {
+      case Some(cache) =>
+        cache.answers
+          .map(answers => Ok(choicePage(Choice.form().fill(Choice(answers.`type`)))))
+          .getOrElse(Ok(choicePage(Choice.form())))
+      case None => Ok(choicePage(Choice.form()))
```

Comment:
Isn't the second case redundant? There is a repetition, so I thought maybe it could be unified?

Reply:
Copied but I'll take a look.

Reply:
Done.  By the power of flatmapping - ```  def displayChoiceForm: Action[AnyContent] = authenticate.async { implicit request =>
    cache.findByEori(request.eori).map(_.flatMap(_.answers)).map {
      case Some(answers) => Ok(choicePage(Choice.form().fill(Choice(answers.`type`))))
      case None          => Ok(choicePage(Choice.form()))
    }
  }```